py4DSTEM/process/diffraction/crystal.py

```diff
@@ -14,12 +14,19 @@
 from py4DSTEM.process.diffraction.crystal_viz import plot_ring_pattern
 from py4DSTEM.process.diffraction.utils import Orientation, calc_1D_profile
 
+from importlib.util import find_spec
+
 try:
     from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
     from pymatgen.core.structure import Structure
 except ImportError:
     pass
 
+try:
+    from ase.io import read
+except ImportError:
+    pass
```

Comment:
For an optional dependency that is not needed at the top level of the file, I think a better pattern is to move the import within the function that requires it. That way any import errors get raised at the appropriate moment.

Reply:
I'm going to delete as its now a dependency